### PR TITLE
Update crd doc generator

### LINKED ISSUE #2138: Migrate to crd-ref-docs for the api docs generation
### Is your feature request related to a problem?

https://github.com/ahmetb/gen-crd-api-reference-docs is not maintained anymore, we should switch to use
https://github.com/elastic/crd-ref-docs

### Describe the solution you'd like

https://github.com/ahmetb/gen-crd-api-reference-docs is not maintained anymore, we should switch to use
https://github.com/elastic/crd-ref-docs

### Additional context

_No response_

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

### PR DESCRIPTION
**Pull request by michaelasp**
Switch from https://github.com/ahmetb/gen-crd-api-reference-docs to https://github.com/elastic/crd-ref-docs since the former is now unsupported. crd-ref-docs can generate markdown automatically, making it so we don't have to paste raw html anymore.

Fixed #2138

### CODE REVIEW

**Comment by michaelasp** [comment]
PTAL whenever you get a chance @fedepaol! My only concern is formatting to make sure that we have everything similar to before since we now generate pure markdown. Let me know if this looks good to you :)

File: out.md [hunk 1]
```diff
@@ -0,0 +1,765 @@
+# API Reference
```
Comment (fedepaol) [inline out.md:1]: committed by mistake?

File: website/content/apis/_index.md [hunk 2]
```diff
@@ -2,985 +2,768 @@
...
+| `apiVersion` _string_ | `metallb.io/v1alpha1`
+| `kind` _string_ | `AddressPool`
+| `kind` _string_ | Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds |
+| `apiVersion` _string_ | APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources |
+| `metadata` _[ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v/#objectmeta-v1-meta)_ | Refer to Kubernetes API documentation for fields of `metadata`. |
+| `spec` _[AddressPoolSpec](#addresspoolspec)_ |  |
+| `status` _[AddressPoolStatus](#addresspoolstatus)_ |  |
+
+
+#### AddressPoolList
```
Comment (fedepaol) [inline website/content/apis/_index.md:43]: Let's ignore lists

File: website/content/apis/_index.md [hunk 3]
```diff
@@ -2,985 +2,768 @@
...
+
+| Field | Description |
+| --- | --- |
+| `protocol` _string_ | Protocol can be used to select how the announcement is done. |
+| `addresses` _string array_ | A list of IP address ranges over which MetalLB has authority. You can list multiple ranges in a single pool, they will all share the same settings. Each range can be either a CIDR prefix, or an explicit start-end range of IPs. |
+| `autoAssign` _boolean_ | AutoAssign flag used to prevent MetallB from automatic allocation for a pool. |
+| `bgpAdvertisements` _[BgpAdvertisement](#bgpadvertisement) array_ | When an IP is allocated from this pool, how should it be translated into BGP announcements? |
+
+
+#### AddressPoolStatus
```
Comment (fedepaol) [inline website/content/apis/_index.md:78]: and statuses (for now)

**Comment by fedepaol** [comment]
@michaelasp did you try to generate the site to see the outcome? That can be done by running hugo server (iirc you must match the very same version we use)

File: website/content/apis/_index.md (hunk 2 above)
Comment (fedepaol) [reply website/content/apis/_index.md:43]: This is how you do that:

https://github.com/metallb/frr-k8s/pull/66/commits/21875cda65c2dc47310317d1d62d2f4cd2cddad9#diff-0bd1bfb29c091e48cde3b1490429624af7ed53a468b72e1370d0d236162b09b7R4

**Comment by michaelasp** [comment]
Took a look, the outcome is mostly the same. However we include v1alpha1 apis, and theres some slight formatting differences, the main one being that we can't have nested tables in markdown without raw html. Wdyt? The main difference is the Spec is defined right below the type. Also excluded some more types to be more consistent with the current doc.

Do we want to keep the alpha APIs out of this? If so I can just add a filter for that too. Otherwise these look to be nearly the same as before, minus the nested tables.

File: out.md (hunk 1 above)
Comment (michaelasp) [reply out.md]: Oops, that was from testing.

File: website/content/apis/_index.md (hunk 2 above)
Comment (michaelasp) [reply website/content/apis/_index.md]: Done!

File: website/content/apis/_index.md (hunk 3 above)
Comment (michaelasp) [reply website/content/apis/_index.md]: Done as well :)

**Comment by michaelasp** [comment]
> > Took a look, the outcome is mostly the same. However we include v1alpha1 apis, and theres some slight formatting differences, the main one being that we can't have nested tables in markdown without raw html. Wdyt? The main difference is the Spec is defined right below the type. Also excluded some more types to be more consistent with the current doc.
> > Do we want to keep the alpha APIs out of this? If so I can just add a filter for that too. Otherwise these look to be nearly the same as before, minus the nested tables.
> 
> Yep, good point. Let's remove alpha (and addresspool too, as it's legacy)

Done!

**Comment by fedepaol** [comment]
LGTM thanks! 
Quay seems to be having a bad day.

**Pull request merged**